docs/blog/2022-12-12-Building-OKD-payload.md

```diff
@@ -0,0 +1,265 @@
+
+The pipeline is fairly simple. Take the [build-from-scratch.yaml](https://github.com/okd-project/okd-payload-pipeline/blob/main/pipelines/build-from-scratch.yaml) for example. It has 3 main tasks:
+* Build the base image and the builder image, with which all the payload images will be built
+    * The builder image is a CentOS Stream 9 container image that includes all the dependencies needed to build payload components and is used as the build environment for them
+    * The built binaries are then layered onto a CentOS Stream 9 base image, creating a payload component image.
+    * The base image is shared across all the images in the release payload
+* Build payload images in batches (starting with the ones that don't have any dependencies)
+* Finally, as all OKD payload component images are in the image stream, the OKD release image is in turn built, using the `oc adm release new` command.
+
+```mermaid
+graph TB
+    subgraph pipeline
+    direction LR
+    
+    pipelineRun{{fa:fa-gears PipelineRun}}-- starts --> taskRunBuilderBase{{fa:fa-gears TaskRun - BatchBuild}}
+    
+    taskRunBuilderBase -- runBefore --> taskRun1{{fa:fa-gears BatchBuild#1}};
+    taskRun1 -- runBefore --> taskRun2{{fa:fa-gears BatchBuild#2}};
+    taskRun2 -- runBefore --> payloadRelease{{fa:fa-gears BuildRelease}}
+    
+    taskRunBuilderBase -- builds --> base(fa:fa-cube base)
+    taskRunBuilderBase -- builds --> builder(fa:fa-cube builder)
+    
+    taskRun1 -- builds --> payloadImage101(fa:fa-cube Payload-101)
+    taskRun1 -- builds --> payloadImage102(fa:fa-cube Payload-102) 
+    
+    taskRun2 -- builds --> payloadImage201(fa:fa-cube Payload-201) 
+    taskRun2 -- builds --> payloadImage202(fa:fa-cube Payload-201)
+    
+    payloadRelease -- builds --> payloadReleaseImage(fa:fa-cube Payload-Release)
+    
+    base                 -- pushed --> IS[(IS origin)]
+    builder              -- pushed --> IS
+    payloadImage101      -- pushed --> IS
+    payloadImage102      -- pushed --> IS
+    payloadImage201      -- pushed --> IS
+    payloadImage202      -- pushed --> IS
+    payloadReleaseImage  -- pushed -->quay[(quay.io)];
+    quay <-.- IS
+    
+    end
+
+```
+
+### Triggers
+
+For the moment, this pipeline has no triggers. It can be executed manually when needed. We are planning to automatically trigger the pipeline on a daily cadence.
+
+### Batch Build Task
+
+With a set of buildConfigs passed in the parameters, this task relies on an openshift `oc` image containing the client binary and loops on the list of build configs with a `oc start-build`, and waits for all the builds to complete.  
```

Comment:
Lets add "other methods of building images - `docker build`, Shipwright - may also be implemented (maybe worth moving to the end of the post?)

Reply:
Maybe between "### Preparing for a new release" and "### Takeaway"?

Reply:
Done. Sorry missed this one. 
PTAL